### ISSUE TITLE
read_pdf fails on specific pdf locally, not through hosted api

### ISSUE THREAD

**Issue by Ianpwest** [nlmatics/llmsherpa#63]
PDF in question: 
[JTR.pdf](https://github.com/nlmatics/llmsherpa/files/14762598/JTR.pdf)

This api call works great
`llmsherpa_api_url = "https://readers.llmsherpa.com/api/document/developer/parseDocument?renderFormat=all"
pdf_url = "JTR.pdf"
pdf_reader = LayoutPDFReader(llmsherpa_api_url)
doc = pdf_reader.read_pdf(pdf_url)`

This local call fails
`llmsherpa_api_url = "[http://localhost:5010/api/parseDocument?renderFormat=all"](http://localhost:5010/api/parseDocument?renderFormat=all%22)
pdf_url = "JTR.pdf"
pdf_reader = LayoutPDFReader(llmsherpa_api_url)
doc = pdf_reader.read_pdf(pdf_url)`

The local version is running from the latest docker build. Other pdfs work fine. Is there a way to get a better error message? Currently receiving: KeyError: 'return_dict'

I noticed there are other issues open around this error but did not find any matching this case where it works on one and not the other. 

I appreciate your time and any insight. Thanks!

**Comment by wolfassi123** [comment]
Hey, @Ianpwest  did you manage to solve this?

**Comment by Ianpwest** [comment]
> Hey, @Ianpwest did you manage to solve this?

@wolfassi123 No, there were also some other parsing issues with different character sets. The library is promising but seemingly under supported. No movement on my tickets.

**Comment by dalmia** [comment]
Switching to the docker image for nlm-ingestor in https://github.com/nlmatics/nlm-ingestor/issues/72#issuecomment-2218239935 comment worked for me.
